modules/packages/Sort.chpl

```diff
@@ -47,24 +198,27 @@ private inline proc chpl_sort_cmp(a, b, param reverse=false, param eq=false) {
    :type doublecheck: `bool`
    :arg reverse: Sort in reverse numerical order
    :type reverse: `bool`
+   :arg comparator: Record that redefines the comparison mechanism with one of
+      the methods: ``comparator.key(a)`` or ``comparator.compare(a,b)``
 
  */
-proc BubbleSort(Data: [?Dom] ?eltType, doublecheck=false, param reverse=false) where Dom.rank == 1 {
+proc BubbleSort(Data: [?Dom] ?eltType, doublecheck=false, param reverse=false, comparator:?rec=emptyRecord) where Dom.rank == 1 && isRecord(rec) {
```

Comment:
Also, I wouldn't put isRecord(rec) in the where clause. Let it be an error if chpl_check_comparator doesn't work. Nothing about how you are using the comparator needs it to be a record (it could be a class, or maybe somebody juts made methods on int, etc)